app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsViewModel.kt

```diff
@@ -119,4 +139,37 @@ class AgendaDetailsViewModel @Inject constructor() : ViewModel() {
             )
         }
     }
+
+    fun onAddPhotosClick(photoUris: List<Uri>) {
+        _photosUri.value = photoUris
+    }
+
+    fun compressAndAddImage(context: Context, uris: List<Uri>) {
+        viewModelScope.launch {
+            var skipped = 0
+            val newCompressedList = uris.mapNotNull { uri ->
+                val drawable = imageCompressionUseCase.uriToDrawable(context, uri)
+                val originalBitmap = (drawable as BitmapDrawable).bitmap
+                Log.d(TAG, "Original Bitmap Size in bytes: ${originalBitmap.byteCount}")
+
+                val compressedByteArray = withContext(Dispatchers.IO) {
+                    imageCompressionUseCase.compressImage(drawable)
+                }
+
+                val compressedBitmap =
+                    BitmapFactory.decodeByteArray(compressedByteArray, 0, compressedByteArray.size)
```

Comment:
This is also a blocking call which will block the main thread here, since you're running it on Dispatchers.Main